src/Rings/MPolyQuo.jl

```diff
@@ -696,13 +702,15 @@ x^3 + x
 ```
 """
 function simplify(f::MPolyQuoRingElem)
+  f.simplified && return f
   R  = parent(f)
   OR = oscar_origin_ring(R)
   SR = singular_origin_ring(R)
   G  = singular_origin_groebner_basis(R)
   g  = f.f
   f.f = OR(reduce(SR(g), G))
-  return f::elem_type(R)
+  f.simplified = true
+  return f::typeof(f)
```

Comment:
Isn't this 100% equivalent to `return f`?

Reply:
Ah yes, not sure what I was thinking. There was some type annotation before and I thought I could improve it, but it makes obviously no sense.